src/main/java/elements/Teleporter.java

```diff
@@ -0,0 +1,18 @@
+package elements;
+import game.Player;
+import game.PlayerState;
+import utils.Point2D;
+
+public class Teleporter extends Element implements Interactable, Teleportable{
+
+    public Teleporter (String sprite, Point2D pos) {
+        super(sprite, pos);
+    }
+
+    @Override
+    public PlayerState changePlayerState(PlayerState playerState) {
+        return playerState;
```

Comment:
If you aren't modifying the player state, you don't need to have this class implement interactable.

Reply:
Removed Interactable implementation

Reply:
Read my other comments. If you choose to implement the changes I suggested, you will need to implement interactable.